iib/workers/tasks/opm_operations.py

```diff
@@ -190,3 +191,216 @@ def _serve_cmd_at_port(serve_cmd, cwd, port, max_tries, wait_time):
+        # https://github.com/operator-framework/enhancements/blob/master/enhancements/substitutes-for.md
+        '--enable-alpha',
+        '--bundle-images',
+        bundle_str,
+    ]
+
+    if container_tool:
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+
+    log.info('Generating the database file with the following bundle(s): %s', ', '.join(bundles))
+
+    if overwrite_csv:
+        log.info('Using force to add bundle(s) to index')
+        cmd.extend(['--overwrite-latest'])
+
+    with set_registry_token(overwrite_from_index_token, from_index):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to add the bundles to the index image')
+
+
+@retry(exceptions=IIBError, tries=2, logger=log)
+def opm_registry_add_fbc(
+    base_dir,
+    bundles,
+    binary_image,
+    from_index=None,
+    overwrite_from_index_token=None,
+    overwrite_csv=False,
+    container_tool=None,
+):
+    """
+    Add the input bundles to an operator index.
+
+    This only produces the index.Dockerfile file and does not build the container image.
+
+    :param str base_dir: the base directory to generate the database and index.Dockerfile in.
+    :param list bundles: a list of strings representing the pull specifications of the bundles to
+        add to the index image being built.
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from. This should point to a digest or stable tag.
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param str overwrite_from_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_from_index``.
+        The format of the token must be in the format "user:password".
+    :param bool overwrite_csv: a boolean determining if a bundle will be replaced if the CSV
+        already exists.
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    if from_index:
+        log.info('Using the existing database from %s', from_index)
+        index_db_file = get_hidden_index_database(from_index, base_dir)
+    else:
+        log.info('Creating new database file index.db')
+        index_db_file = os.path.join(base_dir, get_worker_config()['temp_index_db_path'])
+        index_db_dir = os.path.dirname(index_db_file)
+        if not os.path.exists(index_db_dir):
+            os.makedirs(index_db_dir, exist_ok=True)
+        with open(index_db_file, 'w'):
+            pass
+
+    _opm_registry_add(
+        base_dir=base_dir,
+        index_db=index_db_file,
+        bundles=bundles,
+        from_index=from_index,
+        overwrite_from_index_token=overwrite_from_index_token,
+        overwrite_csv=overwrite_csv,
+        container_tool=container_tool,
+    )
+
+    fbc_dir = opm_migrate(index_db_file, base_dir)
```

Comment:
This is what I meant in my earlier comment actually, within this function opm_migrate() is called with positional arguments whereas opm_generate_dockerfile() is called with named arguments. The reason behind using named arguments sounds good to me, but it would be better to have it consistent. Regardless, this was a small suggestion, should not block merging this PR. feel free to merge like this